### DISCUSSION TITLE
Row groups as new leftmost column in stub

### DISCUSSION THREAD

**Discussion by DaniMori** [rstudio/gt#922] [Q&A] [answered]
Hi, I'd like to know if this is currently doable in `gt`. So far I haven't been able to achieve it and I haven't seen any issue requesting it either, but I wanted to make sure before opening an issue myself.

Mind the following example, taken from [`gt`'s Intro page](https://gt.rstudio.com/articles/intro-creating-gt-tables.html) itself:

![sample table with column of row groups](https://user-images.githubusercontent.com/8235380/161918076-54d82385-27cb-43c1-95c7-e4a469456678.png)

In this example, you can see that the row groups "incoming" and "outgoing" are in a new column left of the row labels in the stub. This allows making the table somehow wider and shorter, thus saving some vertical space (which in some applications can be very convenient).

My question is: is it possible to do this currently in `gt`? My impression is that it's not (also, I believe it would probably require redefining some properties like the "Stubhead" element to take into account the possibility of two [or more?] columns in the stub).

Has anybody been able to achieve this by any sort of workaround, and if so, would you be so kind as to share your code?

Many thanks!

**Comment by bryanwhiting** [comment]
+1 to this idea. This is a basic property of an excel pivot table. `gt` becomes very hard to read when you group by 2+ columns. 

Even grouping by one column can be hard to read because the column header isn't preserved. If you don't have really meaningful labels then it's lost on the reader.

**Reply by DaniMori** [reply]
I'll definitely ask for this in an issue

**Comment by DaniMori** [comment, marked as answer]
Oh... seems like this is already implemented in #577!

**Reply by bryanwhiting** [reply]
Amazing! Dreams come true.
